packages/asset-swapper/src/utils/market_operation_utils/types.ts

```diff
@@ -265,7 +277,8 @@ export interface BatchedOperation<TResult> {
     handleCallResultsAsync(contract: ERC20BridgeSamplerContract, callResults: string): Promise<TResult>;
 }
 
-export interface SourceQuoteOperation<TFillData extends FillData = FillData> extends BatchedOperation<BigNumber[]> {
+export interface SourceQuoteOperation<TFillData extends FillData = FillData, TQuoteData extends QuoteData = BigNumber>
```

Comment:
We should just make it so *all* sources return `BancorQuoteData` (but just called `QuoteData`) in `handleCallResultAsync()`. This one edge case makes the code confusing.

Reply:
refactored